UnityGLTF/Assets/UnityGLTF/Runtime/Scripts/SceneImporter/ImporterMaterials.cs

```diff
@@ -197,15 +229,19 @@ protected virtual async Task ConstructMaterial(GLTFMaterial def, int materialInd
 					var ext = GetTextureTransform(pbr.BaseColorTexture);
 					if (ext != null)
 					{
-						var offset = ext.Offset.ToUnityVector2Raw();
-						offset.y = 1 - ext.Scale.Y - offset.y;
+						CalculateYOffsetAndScale(textureId, ext, out var scale, out var offset);
 						unlitMapper.BaseColorXOffset = offset;
 						unlitMapper.BaseColorXRotation = ext.Rotation;
-						unlitMapper.BaseColorXScale = ext.Scale.ToUnityVector2Raw();
+						unlitMapper.BaseColorXScale = scale;
 						unlitMapper.BaseColorXTexCoord = ext.TexCoord;
 
 						unlitMapper.Material.SetKeyword("_TEXTURE_TRANSFORM", true);
 					}
+					else
+					if (IsTextureFlipped(textureId.Value))
+					{
+						unlitMapper.BaseColorXScale *= new Vector2(1f,-1f);
```

Comment:
I'm not sure if this is the right place here - and if this works with transformed texture coordinates. When is this needed and why is it in the "else" path?